File: scripts/tools/preview_english_verse.ts
import { createClient } from "https://esm.sh/@supabase/supabase-js@2";

const GEMINI_API_KEY = Deno.env.get("GEMINI_API_KEY") || "";

async function generatePreview() {
    const chapter = 1;
    const verse = 1;
    
    const csvPath = "/Users/sandeep/DailyShlokyaAG/scripts/data/mahabharat/chapter_1_source.csv";
    const csvText = await Deno.readTextFile(csvPath);
    const lines = csvText.split('\n');
    const header = lines[1].trim().split(';');
    const dataRows = lines.slice(2);
    
    let targetRow = null;
    for (const line of dataRows) {
        if (!line.trim()) continue;
        const parts = line.split(';');
        if (parts[1] === String(chapter) && parts[2] === String(verse)) {
            targetRow = {};
            header.forEach((key, i) => { targetRow[key] = parts[i]; });
            break;
        }
    }

    if (!targetRow) {
        console.error("Error: Could not find Verse 1.1 in source CSV.");
        return;
    }

    const promptTemplate = await Deno.readTextFile("/Users/sandeep/DailyShlokyaAG/scripts/mahabharat_prompt.md");
    const baseSource = `EN Story: ${targetRow['translation']}\nEN Commentary: ${targetRow['commentary']}\nEN Daily Life: ${targetRow['daily_life_application']}`;
    
    const prompt = promptTemplate.replace("{{chapter}}", String(chapter))
                           .replace("{{verse}}", String(verse))
                           .replace("{{base_source}}", baseSource);

    console.log(`--- Generating English Preview for Verse ${chapter}.${verse} ---`);
    
    const resp = await fetch(`https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:generateContent?key=${GEMINI_API_KEY}`, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({
          contents: [{ parts: [{ text: prompt }] }],
          generationConfig: { response_mime_type: "application/json" },
        }),
    });

    if (!resp.ok) {
        console.error(`Gemini Error: ${await resp.text()}`);
        return;
    }

    const data = await resp.json();
    const content = JSON.parse(data.candidates[0].content.parts[0].text);
    const en = content.en;
    
    console.log(`\nTITLE: ${en.title}`);
    console.log(`\nSTORY (Length: ${en.translation.split(/\s+/).length} words):\n`);
    console.log(en.translation.replace(/<break time="500ms"\/>/g, "..."));
    console.log(`\nCOMMENTARY:\n${en.commentary}`);
    console.log(`\nDAILY LIFE APPLICATION:\n${en.daily_life_application}`);
    console.log(`\nPRACTICAL EXAMPLES:\n1. ${en.practical_examples[0]}\n2. ${en.practical_examples[1]}`);
}

generatePreview();

File: scripts/tools/preview_english_verse.py
import google.generativeai as genai
import os
import json
import csv

# Setup Gemini
genai.configure(api_key=os.environ.get("GEMINI_API_KEY"))
model = genai.GenerativeModel('gemini-1.5-flash')

def get_csv_row(chapter, verse):
    csv_path = "/Users/sandeep/DailyShlokyaAG/scripts/data/mahabharat/chapter_1_source.csv"
    with open(csv_path, mode='r', encoding='utf-8') as f:
        # The CSV has some metadata on line 1, header on line 2
        lines = f.readlines()
        header = lines[1].strip().split(';')
        data_rows = lines[2:]
        reader = csv.DictReader(data_rows, fieldnames=header, delimiter=';')
        for row in reader:
            if row['chapter_no'] == str(chapter) and row['verse_no'] == str(verse):
                return row
    return None

def generate_preview():
    chapter = 1
    verse = 1
    row = get_csv_row(chapter, verse)
    if not row:
        print("Error: Could not find Verse 1.1 in source CSV.")
        return

    # Load Prompt Template
    with open("/Users/sandeep/DailyShlokyaAG/scripts/mahabharat_prompt.md", "r") as f:
        prompt_template = f.read()

    base_source = f"EN Story: {row['translation']}\nEN Commentary: {row['commentary']}\nEN Daily Life: {row['daily_life_application']}"
    
    prompt = prompt_template.replace("{{chapter}}", str(chapter))\
                           .replace("{{verse}}", str(verse))\
                           .replace("{{base_source}}", base_source)

    print(f"--- Generating English Preview for Verse {chapter}.{verse} ---")
    response = model.generate_content(
        prompt,
        generation_config={"response_mime_type": "application/json"}
    )
    
    content = json.loads(response.text)
    en = content['en']
    
    print(f"\nTITLE: {en['title']}")
    print(f"\nSTORY (Length: {len(en['translation'].split())} words):\n")
    print(en['translation'])
    print(f"\nCOMMENTARY:\n{en['commentary']}")
    print(f"\nDAILY LIFE APPLICATION:\n{en['daily_life_application']}")
    print(f"\nPRACTICAL EXAMPLES:\n1. {en['practical_examples'][0]}\n2. {en['practical_examples'][1]}")

if __name__ == "__main__":
    generate_preview()
